x-pack/plugins/fleet/server/services/epm/packages/_install_package.ts

```diff
@@ -214,6 +217,15 @@ export async function _installPackage({
       logger.warn(`Error removing legacy templates: ${e.message}`);
     }
 
+    const { diagnosticFileUploadEnabled } = appContextService.getExperimentalFeatures();
```

Comment:
We check the integration has file upload in the function, so it will exit very early here: https://github.com/hop-dev/kibana/blob/203833e5ca62ad46d727b4222874bfd11edf01f7/x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts#L375 Sp the performance impact will just be one array lookup.

We need to run on setup to add the indices for packages already installed in previous kibana versions, we have to run on install for new package installs.